DMCompiler/DMStandard/DefaultPragmaConfig.dm

```diff
@@ -32,6 +32,7 @@
 #pragma SuspiciousSwitchCase warning
 #pragma PointlessPositionalArgument warning
 #pragma ProcArgumentGlobal warning // Ref BYOND issue https://www.byond.com/forum/post/2830750
+#pragma AmbiguousVarStatic error // https://github.com/OpenDreamProject/OpenDream/issues/997
```

Comment:
I think this should be a notice by default. When a static of the same name exists it's fine to use that.

When I said it should be an error, I was thinking of situations where there was only an instance variable with the name.
```suggestion
#pragma AmbiguousVarStatic notice // https://github.com/OpenDreamProject/OpenDream/issues/997
```

Reply:
I think if a codebase needs to use ambiguous names like this, they could do this on their own. Otherwise it prevents awful footguns.

The thing didn't trigger on Paradise at all. Not sure about other codebases though.

Reply:
The general philosophy with these pragmas has been to not error on things that byond doesn't. If a codebase wants the extra guarantee of erroring on it, they can raise it themselves (as they've been doing up to now).

I said `notice` because I felt it was a little pedantic of a lint, but I can see a case being made for `warning`.

Reply:
Ah, I see. 

Probably want a pedantic mode in compiler that uses harsher pragmas, eventually

Probably should be a warning. It's gonna be pretty ambiguous when trying to read it unless you keep this behavior in mind.